pkg/authorization/registry/subjectaccessreview/rest.go

```diff
@@ -34,12 +35,25 @@ func (r *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, err
 		return nil, fmt.Errorf("not a subjectAccessReview: %#v", obj)
 	}
 
-	namespace := kapi.NamespaceValue(ctx)
-	user := &user.DefaultInfo{
-		Name:   subjectAccessReview.User,
-		Groups: subjectAccessReview.Groups,
+	var userToCheck user.Info
+	if (len(subjectAccessReview.User) == 0) && (len(subjectAccessReview.Groups) == 0) {
+		// if no user or group was specified, use the info from the context
+		exists := false
+		userToCheck, exists = kapi.UserFrom(ctx)
```

Comment:
Because that's what our style guide requires?

----- Original Message -----
> > @@ -34,12 +35,25 @@ func (r *REST) Create(ctx kapi.Context, obj
> > runtime.Object) (runtime.Object, err
> >  		return nil, fmt.Errorf("not a subjectAccessReview: %#v", obj)
> >  	}
> >  
> > -	namespace := kapi.NamespaceValue(ctx)
> > -	user := &user.DefaultInfo{
> > -		Name:   subjectAccessReview.User,
> > -		Groups: subjectAccessReview.Groups,
> > +	var userToCheck user.Info
> > +	if (len(subjectAccessReview.User) == 0) &&
> > (len(subjectAccessReview.Groups) == 0) {
> > +		// if no user or group was specified, use the info from the context
> > +		exists := false
> > +		userToCheck, exists = kapi.UserFrom(ctx)
> 
> beauty is in the eye of the beholder or is there some real reason?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/1186/files#r25805499